render/egl.c

```diff
@@ -172,6 +172,7 @@ struct wlr_egl *wlr_egl_create_with_drm_fd(int drm_fd) {
 		}
 		return NULL;
 	}
+	wlr_log(WLR_INFO, "Supported EGL client extensions: %s", client_exts_str);
 
 	if (!check_egl_ext(client_exts_str, "EGL_KHR_platform_gbm")) {
```

Comment:
If we use `EGL_PLATFORM_DEVICE_EXT`, then we don't need the GBM platform. We should move this check to `egl_init` and guard it behind a `platform` check.

Reply:
I added a bool in `wlr_egl` since `egl_init` doesn't have access to the client exts string. I also move the platform check on top of the function to not initialize too many EGL stuff for nothing.

Reply:
This comment still needs to be addressed. If an EGL impl doesn't have `EGL_KHR_platform_gbm` but does have `EGL_EXT_platform_device`, we should be able to initialize EGL with an `EGLDevice`.